### PR TITLE
feat(admin-ui): improve facet selector with the code

### PR DESCRIPTION
**Pull request by alexisvigoureux**
# Description

Sometimes multiple facets have the same name so it's impossible to distinguish them except with code name
This PR introduces the code name in the facet value chip component

PS: I have not found the way to generate the graphql schema for the admin-ui directory

# Breaking changes

No

## Test

Add a facet with the same name (but different code) as another one and with the same facetValues
Go to http://localhost:4200/admin/catalog/collections/2
Add the facet in the collection filters

# Screenshots

Add a popover to include more context about the facet type
Capture d’écran 2024-10-31 à 19 16 49

Update the title displayed on hover to include more context about the facet type
Capture d’écran 2024-10-31 à 19 17 06

# Checklist

📌 Always:
- [x] I have set a clear title
- [x] My PR is small and contains a single feature
- [x] I have [checked my own PR](## "Fix typo's and remove unused or commented out code")

👍 Most of the time:
- [ ] I have added or updated test cases
- [ ] I have updated the README if needed

### CODE REVIEW

File: packages/admin-ui/src/lib/core/src/shared/components/facet-value-chip/facet-value-chip.component.ts [hunk 1]
```diff
@@ -13,4 +13,9 @@ export class FacetValueChipComponent {
     @Input() removable = true;
     @Input() displayFacetName = true;
     @Output() remove = new EventEmitter<void>();
+
+    get formattedTitle(): string {
+        const facetCode = this.facetValue.facet.code ? `(${this.facetValue.facet.code}) ` : '';
```
Comment (michaelbromley) [inline packages/admin-ui/src/lib/core/src/shared/components/facet-value-chip/facet-value-chip.component.ts]: What's the need for this line? Do you expect that the `facet.code` may sometimes be falsy?

File: packages/admin-ui/src/lib/core/src/shared/components/facet-value-chip/facet-value-chip.component.ts [hunk 2]
```diff
@@ -13,4 +13,9 @@ export class FacetValueChipComponent {
     @Input() removable = true;
     @Input() displayFacetName = true;
     @Output() remove = new EventEmitter<void>();
+
+    get formattedTitle(): string {
```
Comment (michaelbromley) [inline packages/admin-ui/src/lib/core/src/shared/components/facet-value-chip/facet-value-chip.component.ts]: One note: using getters in template bindings can result in _lots_ of calls. Try adding a console log in this getter and see. If it is resulting in excessive calls, consider saving the title to a regular class field and setting in in the `OnInit` lifecycle hook.

**Review by michaelbromley — COMMENTED** [review]
Thanks for the improvement! I've left a couple of comments I'd like you to look at.

File: packages/admin-ui/src/lib/core/src/shared/components/facet-value-chip/facet-value-chip.component.ts (hunk 2 above)
Comment (alexisvigoureux) [reply packages/admin-ui/src/lib/core/src/shared/components/facet-value-chip/facet-value-chip.component.ts]: https://github.com/vendure-ecommerce/vendure/pull/3175/commits/b499e03717164d0a0556f5edcb97aaebf0a80f4f

File: packages/admin-ui/src/lib/core/src/shared/components/facet-value-chip/facet-value-chip.component.ts (hunk 1 above)
Comment (alexisvigoureux) [reply packages/admin-ui/src/lib/core/src/shared/components/facet-value-chip/facet-value-chip.component.ts]: I expect the code to be empty (I've updated the code to reflect this) if it's not requested in the graphql request
https://github.com/vendure-ecommerce/vendure/pull/3175/commits/b499e03717164d0a0556f5edcb97aaebf0a80f4f (example: the product page)

**Comment by michaelbromley** [comment]
Thank you!

**Pull request merged**